modules/vmseries/variables.tf

```diff
@@ -1,219 +1,232 @@
+                                      used to encrypt this VM's disk
+  - `encryption_at_host_enabled`    - (`bool`, optional, defaults to Azure defaults) should all of disks be encrypted
+                                      by enabling Encryption at Host
+  - `proximity_placement_group_id`  - (`string`, optional, defaults to Azure defaults) the ID of the Proximity Placement Group
+                                      in which the Firewall should be assigned to
+  - `diagnostics_storage_uri`       - (`string`, optional, defaults to `null`) storage account's blob endpoint to hold
+                                      diagnostic files
+  - `identity_type`                 - (`string`, optional, defaults to `SystemAssigned`) type of Managed Service Identity that
+                                      should be configured on this VM. Can be one of "SystemAssigned", "UserAssigned" or
+                                      "SystemAssigned, UserAssigned".
+  - `identity_ids`                  - (`list`, optional, defaults to `[]`) a list of User Assigned Managed Identity IDs to be 
+                                      assigned to this VM. Required only if `identity_type` is not "SystemAssigned"
 
-variable "avset_id" {
-  description = "The identifier of the Availability Set to use. When using this variable, set `avzone = null`."
-  default     = null
-  type        = string
+  EOF
+  type = object({
+    size                         = optional(string, "Standard_D3_v2")
+    bootstrap_options            = optional(string)
+    zone                         = string
+    disk_type                    = optional(string, "StandardSSD_LRS")
+    disk_name                    = string
+    avset_id                     = optional(string)
+    accelerated_networking       = optional(bool, true)
+    encryption_at_host_enabled   = optional(bool)
+    proximity_placement_group_id = optional(string)
+    disk_encryption_set_id       = optional(string)
+    diagnostics_storage_uri      = optional(string)
+    identity_type                = optional(string, "SystemAssigned")
+    identity_ids                 = optional(list(string), [])
+  })
+  validation {
+    condition     = contains(["Standard_LRS", "StandardSSD_LRS", "Premium_LRS"], var.virtual_machine.disk_type)
+    error_message = "The `disk_type` property can be one of: `Standard_LRS`, `StandardSSD_LRS` or `Premium_LRS`."
+  }
+  validation {
+    condition     = contains(["SystemAssigned", "UserAssigned", "SystemAssigned, UserAssigned"], var.virtual_machine.identity_type)
+    error_message = "The `identity_type` property can be one of \"SystemAssigned\", \"UserAssigned\" or \"SystemAssigned, UserAssigned\"."
+  }
+  validation {
+    condition     = var.virtual_machine.identity_type == "SystemAssigned" ? length(var.virtual_machine.identity_ids) == 0 : length(var.virtual_machine.identity_ids) >= 0
```

Comment:
In false case, shouldn't `var.virtual_machine.identity_ids` length be higher than 0 rather than higher or equal?